src/bridge/cockpitcpusamples.c

```diff
@@ -112,3 +112,76 @@ cockpit_cpu_samples (CockpitSamples *samples)
   g_strfreev (lines);
   g_free (contents);
 }
+
+gchar*
+read_hwmon_file (gchar *path, GError **error)
+{
+  gchar *content = NULL;
+
+  if (!g_file_get_contents (path, &content, NULL, error))
+    {
+      if (!g_error_matches (*error, G_FILE_ERROR, G_FILE_ERROR_NOENT))
+          g_message ("error loading name: %s", (*error)->message);
```

Comment:
So more GLib conventions time.

It's customary that `error` here might be `NULL`, and even if we know that it never is, you should write the function in a way that it might be.

The way that usually works is by using a `g_autoptr(GError) my_error` and then checking the condition that you want to check.  In case you want to re-raise the error.  That looks like `g_propagate_error (error, g_steal_pointer (&my_error));`.  Otherwise, just do nothing, and the error gets freed automatically.

Which brings to the next point: what do you intend to do differently if you get `NOENT` here?  There are a couple of options.  Although it's slightly non-idiomatic, I feel like it would be okayish to return `NULL` here in case of `ENOENT`, but not propagate the error.  In case of "hard" errors, you would propagate.

If you want to be hardcore idiomatic (ie: this is "proper GLib-style API"), you would probably write a function with a signature like

```
gboolean
read_hwmon_file (const gchar *path,
                 gchar **contents,
                 GError **error);
```

which would perform as follows:

 - successfully read file: set `*contents` and return `TRUE`
 - `ENOENT`: set `*contents = NULL` and return `TRUE`
 - other error: propagate error, `*contents` undefined, return `FALSE`.